### PR TITLE
Add support for NXP I2C NTAG

### PR DESCRIPTION
**Pull request by fredlee12001**

### CODE REVIEW

File: NFC_EEPROM/source/target/TARGET_NT3H2111/EEPROMDriver.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "EEPROMDriver.h"
+#include "nt3h_driver.h"
+
+mbed::nfc::NFCEEPROMDriver& get_eeprom_driver(events::EventQueue&)
+{
+    static mbed::nfc::vendor::NXP::NT3HDriver eeprom_driver(D14,D15,D5);
```
Comment (karsev) [inline NFC_EEPROM/source/target/TARGET_NT3H2111/EEPROMDriver.cpp]: A more generic issue, should there be generic names for the pin configurations? 
E.g. NFC_I2C_SDA etc., which can then be configured in the application .json. And if not defined then those pins are used.

**Comment by dlfryar** [comment]
cc: @maclobdell

File: NFC_EEPROM/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ You will also need to supply the driver for the EEPROM. This example is known to
 Clone the repository containing the collection of examples:
 
 ```
-git clone https://github.com/ARMmbed/mbed-os-example-nfc.git
+git clone https://github.com/vince-zeng/mbed-os-example-nfc.git
```
Comment (paul-szczepanek-arm) [inline NFC_EEPROM/README.md]: this needs to point back to arm repo

**Comment by paul-szczepanek-arm** [comment]
Did you run the NFC tests on the chip? (mbed test -greentea -n TESTS-nfc-eeprom)

File: NFC_EEPROM/README.md (hunk 2 above)
Comment (fredlee12001) [reply NFC_EEPROM/README.md]: changed

File: NFC_EEPROM/source/target/TARGET_NT3H2111/EEPROMDriver.cpp (hunk 1 above)
Comment (fredlee12001) [reply NFC_EEPROM/source/target/TARGET_NT3H2111/EEPROMDriver.cpp]: Changed to use mbed config macro now.

**Comment by fredlee12001** [comment]
Hi @paul-szczepanek-arm , 

I have fixed the errors and passed all the NFC greentea tests now.  Would you kindly help review that?

Thanks.

mbedgt: test suite report:
| target       | platform_name | test suite               | result | elapsed_time (sec) | copy_method |
|--------------|---------------|--------------------------|--------|--------------------|-------------|
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | OK     | 21.58              | default     |
mbedgt: test suite results: 1 OK
mbedgt: test case report:
| target       | platform_name | test suite               | test case                       | passed | failed | result | elapsed_time (sec) |
|--------------|---------------|--------------------------|---------------------------------|--------|--------|--------|--------------------|
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-ERASE-BYTES           | 1      | 0      | OK     | 0.09               |
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-SESSION               | 1      | 0      | OK     | 0.04               |
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-TRUNCATE-SIZE         | 1      | 0      | OK     | 0.11               |
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-WRITE-BEYOND-SIZE     | 1      | 0      | OK     | 0.1                |
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-WRITE-BYTES-TRUNCATED | 1      | 0      | OK     | 0.1                |
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-WRITE-READ            | 1      | 0      | OK     | 0.1                |
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-WRITE-READ-SIZE       | 1      | 0      | OK     | 0.08               |
| K64F-GCC_ARM | K64F          | mbed-os-tests-nfc-eeprom | NFCEEPROM-WRITE-WITH-OFFSET     | 1      | 0      | OK     | 0.11               |
mbedgt: test case results: 8 OK
mbedgt: completed in 21.73 sec

**Comment by paul-szczepanek-arm** [comment]
conflict hard to resolve, replaces by https://github.com/ARMmbed/mbed-os-example-nfc/pull/92

**Pull request closed without merging**